be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -40,10 +40,41 @@ class RollCallHandler(dbRef: => AskableActorRef) extends MessageHandler {
   private val unknownAnswer: String = "Database actor returned an unknown answer"
 
   def handleCreateRollCall(rpcRequest: JsonRpcRequest): GraphMessage = {
-    val ask: Future[GraphMessage] = dbAskWritePropagate(rpcRequest)
-    Await.result(ask, duration)
+    rpcRequest.getParamsMessage match {
+      case Some(message: Message) =>
+        val data: CreateRollCall = message.decodedData.get.asInstanceOf[CreateRollCall]
+        // we are using the rollcall id instead of the message_id at rollcall creation
+        val rollCallChannel: Channel = Channel(s"${Channel.ROOT_CHANNEL_PREFIX}${data.id}")
+        val ask =
+          for (
+            channelExists <- dbActor ? DbActor.ChannelExists(rollCallChannel)
+          ) yield channelExists match {
+            case DbActorAck =>
```

Comment:
```suggestion
            case DbActor.DbActorAck =>
```
Does not compile on my end